src/NuGet.Services.Staging.Manager/Controllers/PackageController.cs

```diff
@@ -120,16 +123,20 @@ public class PackageController : Controller
                
                 var packageLocations = await v3Service.AddPackage(packageStream, packageMetadata);
 
-                await _stageService.AddPackageToStage(stage, new StagedPackage
-                {
-                    Id = registrationId,
-                    NormalizedVersion = normalizedVersion,
-                    Version = version.ToString(),
-                    UserKey = userKey,
-                    Published = DateTime.UtcNow,
-                    NupkgUrl = packageLocations.Nupkg.ToString(),
-                    NuspecUrl = packageLocations.Nuspec.ToString()
-                });
+                await _stageService.AddPackageToStage(
+                    stage,
+                    new StagedPackage
+                    {
+                        Id = registrationId,
+                        NormalizedVersion = normalizedVersion,
+                        Version = version.ToString(),
+                        UserKey = userKey,
+                        Published = DateTime.UtcNow,
+                        NupkgUrl = packageLocations.Nupkg.ToString(),
+                        NuspecUrl = packageLocations.Nuspec.ToString(),
+                        PackageMetadata = new PackageMetadata().LoadFromNuspec(nuspec, GetUserInformation())

```

Comment:
Perhaps make this a `PackageMetadataReader` or something? Or a static method `PackageMetadata.FromNuspec()`